sdk/nestjs/src/RequireVariableValue.ts

```diff
@@ -0,0 +1,69 @@
+import {
+    CanActivate,
+    ExecutionContext,
+    Inject,
+    Type,
+    UseGuards,
+    applyDecorators,
+    mixin,
+} from '@nestjs/common'
+import { DVCVariableValue, DevCycleClient } from '@devcycle/nodejs-server-sdk'
+
+type VariableEvaluation = (val: DVCVariableValue) => boolean
+type VariableValues = {
+    [key: string]: DVCVariableValue | VariableEvaluation
+}
+
+export const RequireVariableValue = (
+    requiredVariables: VariableValues,
+): ClassDecorator & MethodDecorator =>
+    applyDecorators(UseGuards(RequireVariableValueGuard(requiredVariables)))
+
+export const RequireVariableValueGuard = (
+    requiredVariableValues: VariableValues,
+): Type<CanActivate> => {
+    class RequireVariableValueGuard implements CanActivate {
+        constructor(
+            @Inject(DevCycleClient) readonly dvcClient: DevCycleClient,
+        ) {}
+
+        async canActivate(context: ExecutionContext): Promise<boolean> {
+            const request = context.switchToHttp().getRequest()
+            const { dvc_user } = request
+
+            if (!dvc_user) {
+                throw new Error(
+                    'User context not found. Make sure to add the DevCycleUserGuard.',
+                )
+            }
+
+            const servedVariables = this.dvcClient.allVariables(dvc_user)
```

Comment:
could we use a series of variable calls here so evaluations get tracked?

Reply:
I originally tried that, but we would need to pass a default value, which doesn't really make sense in this context